src/Microsoft.AspNet.Mvc.ModelBinding/Binders/MutableObjectModelBinder.cs

```diff
@@ -236,24 +231,37 @@ private static bool CanUpdateReadOnlyProperty(Type propertyType)
             return true;
         }
 
-        private async Task<ModelBindingResult> CreateAndPopulateDto(
+        private async Task<IDictionary<ModelMetadata, ModelBindingResult>> BindProperties(
             ModelBindingContext bindingContext,
             IEnumerable<ModelMetadata> propertyMetadatas)
         {
-            // create a DTO and call into the DTO binder
-            var dto = new ComplexModelDto(bindingContext.ModelMetadata, propertyMetadatas);
-
-            var metadataProvider = bindingContext.OperationBindingContext.MetadataProvider;
-            var dtoMetadata = metadataProvider.GetMetadataForType(typeof(ComplexModelDto));
+            var result = new Dictionary<ModelMetadata, ModelBindingResult>();
+            foreach (var propertyMetadata in propertyMetadatas)
+            {
+                var propertyModelName = ModelBindingHelper.CreatePropertyModelName(
+                        bindingContext.ModelName,
+                        propertyMetadata.BinderModelName ?? propertyMetadata.PropertyName);
 
-            var childContext = ModelBindingContext.GetChildModelBindingContext(
-                bindingContext,
-                bindingContext.ModelName,
-                dtoMetadata);
+                var propertyBindingContext = ModelBindingContext.GetChildModelBindingContext(
+                    bindingContext,
+                    propertyModelName,
+                    propertyMetadata);
 
-            childContext.Model = dto;
+                var modelBindingResult =
+                    await bindingContext.OperationBindingContext.ModelBinder.BindModelAsync(propertyBindingContext);
+                if (modelBindingResult == null)
+                {
+                    // Could not bind. Add a result so MutableObjectModelBinder will check for [DefaultValue].
```

Comment:
nit: you're in `MutableObjectModelBinder` :smiley_cat: think you want to refer to `ProcessResult()`